hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -647,36 +573,22 @@ knownUtxo = \case
   take2' (i, o, _) = (i, o)
 
 -- | Look for the "initial" which corresponds to given cardano verification key.
-ownInitial :: VerificationKey PaymentKey -> [(TxIn StandardCrypto, TxOut Era, Data Era)] -> Maybe (TxIn StandardCrypto, PubKeyHash)
+ownInitial ::
+  VerificationKey PaymentKey ->
+  [(TxIn, TxOut CtxUTxO Era, ScriptData)] ->
+  Maybe (TxIn, Hash PaymentKey)
 ownInitial vkey =
   foldl' go Nothing
  where
   go (Just x) _ = Just x
   go Nothing (i, _, dat) = do
-    pkh <- fromData (getPlutusData dat)
-    guard $ pkh == pubKeyHash vkey
-    pure (i, pkh)
+    let vkh = verificationKeyHash vkey
+    pkh <- fromData (toPlutusData dat)
+    guard $ pkh == toPlutusKeyHash vkh
+    pure (i, vkh)
```

Comment:
Keeping this the `pkh` is less conversions and what we have right now on `master`.